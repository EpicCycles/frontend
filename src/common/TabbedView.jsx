import React from "react";
import * as PropTypes from "prop-types";

// have an array of title, parent component has state which is current selected tab index

const TabbedView  = (props) => {
   const { currentTab, tabs, changeTab  } = props;
   const selectedTab = currentTab;
    return <ul className="tabrow" data-test="tab-row">
        {tabs.map ((tab, index) => <li
            id={`tab${index}`}
            key={`tab${index}`}
            onClick={() => (index !== selectedTab) && changeTab(index)}
            className={(index === selectedTab) ? "tabSelected" : "tabUnselected"}
            data-test="tab"
        >
            {tab}
        </li>)}
    </ul>
};

TabbedView.defaultProps = {
   currentTab: 0,
};
TabbedView.propTypes = {
    currentTab: PropTypes.number,
    tabs: PropTypes.array.isRequired,
    changeTab: PropTypes.func.isRequired,
};
export default TabbedView;