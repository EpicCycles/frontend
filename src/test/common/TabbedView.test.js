import React from 'react';
import toJson from 'enzyme-to-json';
import TabbedView from '../../common/TabbedView';

const tabs = ['Addresses', 'Phone numbers', 'Fittings', 'History', 'Quotes'];
describe('TabbedView', () => {
  test('it selects the first tab if none passed', () => {
    const component = shallow(<TabbedView tabs={tabs} changeTab={jest.fn()} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  test('it selects the passed tab', () => {
    const component = shallow(<TabbedView tabs={tabs} changeTab={jest.fn()} currentTab={3} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  test('it calls the function with the right value', () => {
    const changeTab = jest.fn();
    const component = shallow(<TabbedView tabs={tabs} changeTab={changeTab} currentTab={3} />);
    component.find('#tab1').simulate('click');
    expect(changeTab).toHaveBeenCalledWith(1);
  });
  test('it does not call the function if the tab is selected', () => {
    const changeTab = jest.fn();
    const component = shallow(<TabbedView tabs={tabs} changeTab={changeTab} currentTab={3} />);
    component.find('#tab3').simulate('click');
    expect(changeTab).not.toHaveBeenCalled();
  });
});
