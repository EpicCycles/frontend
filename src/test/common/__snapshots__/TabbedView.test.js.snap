// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedView it selects the first tab if none passed 1`] = `
<ul
  className="tabrow"
  data-test="tab-row"
>
  <li
    className="tabSelected"
    data-test="tab"
    id="tab0"
    key="tab0"
    onClick={[Function]}
  >
    Addresses
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab1"
    key="tab1"
    onClick={[Function]}
  >
    Phone numbers
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab2"
    key="tab2"
    onClick={[Function]}
  >
    Fittings
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab3"
    key="tab3"
    onClick={[Function]}
  >
    History
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab4"
    key="tab4"
    onClick={[Function]}
  >
    Quotes
  </li>
</ul>
`;

exports[`TabbedView it selects the passed tab 1`] = `
<ul
  className="tabrow"
  data-test="tab-row"
>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab0"
    key="tab0"
    onClick={[Function]}
  >
    Addresses
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab1"
    key="tab1"
    onClick={[Function]}
  >
    Phone numbers
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab2"
    key="tab2"
    onClick={[Function]}
  >
    Fittings
  </li>
  <li
    className="tabSelected"
    data-test="tab"
    id="tab3"
    key="tab3"
    onClick={[Function]}
  >
    History
  </li>
  <li
    className="tabUnselected"
    data-test="tab"
    id="tab4"
    key="tab4"
    onClick={[Function]}
  >
    Quotes
  </li>
</ul>
`;
